src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -41,6 +41,7 @@
       Outputs="$(AssemblyInfoFile)"
       Condition="'$(GenerateAssemblyInfo)'=='true'">
 
+    <MakeDir Condition="!Exists('$(AssemblyInfoFileDir)')" Directories="$(AssemblyInfoFileDir)" ContinueOnError="true" />
```

Comment:
It isn't necessary, but it saves the expensive task invocation at the cost of a cheap existence check.